frame/salary/src/lib.rs

```diff
@@ -0,0 +1,347 @@
+		#[pallet::weight(T::WeightInfo::add_member())]
+		#[pallet::call_index(1)]
+		pub fn bump(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let now = frame_system::Pallet::<T>::block_number();
+			let cycle_period = T::RegistrationPeriod::get() + T::PayoutPeriod::get();
+			let status = match Status::<T, I>::get() {
+				// Not first time... (move along start block and bump index)
+				Some(mut status) => {
+					status.cycle_start.saturating_accrue(cycle_period);
+					ensure!(now >= status.cycle_start, Error::<T, I>::NotYet);
+					status.cycle_index.saturating_inc();
+					status.budget = T::Budget::get();
+					status.total_registrations = Zero::zero();
+					status.total_unregistered_paid = Zero::zero();
+					status
+				},
+				// First time... (initialize)
+				None => StatusType {
+					cycle_index: Zero::zero(),
+					cycle_start: now,
+					budget: T::Budget::get(),
+					total_registrations: Zero::zero(),
+					total_unregistered_paid: Zero::zero(),
+				},
+			};
+			Status::<T, I>::put(&status);
+
+			Self::deposit_event(Event::<T, I>::CycleStarted { index: status.cycle_index });
+			Ok(Pays::No.into())
+		}
+
+		/// Register for a payout.
+		///
+		/// Will only work if we are in the first `RegistrationPeriod` blocks since the cycle
+		/// started.
+		///
+		/// - `origin`: A `Signed` origin of an account which is a member of `Members`.
+		#[pallet::weight(T::WeightInfo::add_member())]
+		#[pallet::call_index(2)]
+		pub fn register(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let rank = T::Members::rank_of(&who).ok_or(Error::<T, I>::NotMember)?;
+			let mut status = Status::<T, I>::get().ok_or(Error::<T, I>::NotStarted)?;
+			let mut claimant = Claimant::<T, I>::get(&who).ok_or(Error::<T, I>::NotInducted)?;
+			let now = frame_system::Pallet::<T>::block_number();
+			ensure!(
+				now < status.cycle_start + T::RegistrationPeriod::get(),
+				Error::<T, I>::TooLate
+			);
+			ensure!(claimant.last_active < status.cycle_index, Error::<T, I>::NoClaim);
+			let payout = T::ActiveSalaryForRank::convert(rank);
+			ensure!(!payout.is_zero(), Error::<T, I>::ClaimZero);
+			claimant.last_active = status.cycle_index;
+			claimant.unpaid = Some(payout);
+			status.total_registrations.saturating_accrue(payout);
+
+			Claimant::<T, I>::insert(&who, &claimant);
+			Status::<T, I>::put(&status);
+
+			Self::deposit_event(Event::<T, I>::Registered { who, amount: payout });
+			Ok(Pays::No.into())
+		}
+
+		/// Request a payout.
+		///
+		/// Will only work if we are after the first `RegistrationPeriod` blocks since the cycle
+		/// started but by no more than `PayoutPeriod` blocks.
+		///
+		/// - `origin`: A `Signed` origin of an account which is a member of `Members`.
+		#[pallet::weight(T::WeightInfo::add_member())]
+		#[pallet::call_index(3)]
+		pub fn payout(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
```

Comment:
Could we also add some payout that takes as a second parameter some other account as beneficial?

Reply:
Your wish is my command...